pygac/gac_reader.py

```diff
@@ -256,8 +259,72 @@ def get_calibrated_channels(self):
                 self.scans["scan_line_number"],
                 chan,
                 self.spacecraft_name)
+
+        # Mask out corrupt values
+        mask = self.get_corrupt_mask()
+        channels[mask] = np.nan
+
+        # Apply KLM/POD specific postprocessing
+        self.postproc(channels)
+
+        # Mask pixels affected by scan motor issue
+        if self.is_tsm_affected():
+            LOG.info('Correcting for temporary scan motor issue')
+            self.mask_tsm_pixels(channels)
+
         return channels
 
+    def get_lonlat(self):
+        """Compute lat/lon coordinates.
+
+        TODO: Switch to faster interpolator?
+        """
+        if self.lons is None and self.lats is None:
+            lons, lats = self._get_lonlat()
+
+            # Interpolate from every eigth pixel to all pixels.
+            self.lons, self.lats = gtp.Gac_Lat_Lon_Interpolator(lons, lats)
+
+            # Adjust clock drift
+            self.adjust_clock_drift()
+
+            # Mask out corrupt scanlines
+            mask = self.get_corrupt_mask()
+            self.lons[mask] = np.nan
+            self.lats[mask] = np.nan
+
+            # Mask values outside the valid range
+            self.lats[np.fabs(self.lats) > 90.0] = np.nan
+            self.lons[np.fabs(self.lons) > 180.0] = np.nan
+
+        return self.lons, self.lats
+
+    @abstractmethod
+    def _get_lonlat(self):
+        """KLM/POD specific readout of lat/lon coordinates."""
+        raise NotImplementedError
+
+    def get_corrupt_mask(self):
+        """Get mask for corrupt scanlines."""
+        if self.mask is None:
+            self.mask = self._get_corrupt_mask()
+        return self.mask
```

Comment:
Just an idea: should we make `self.mask` a property calling this function ? the current `self.mask` used for caching could be renamed instead `self._mask`

Reply:
Good idea! Probably the same for `self.utcs`, `self.lons` and `self.lats`, right?

Reply:
yes, why not

Reply:
although I only see mask being used all the time in the code here, so maybe the others don't need it, I don't know

Reply:
I made only `self.mask` a property. There is too much in-place modification of `selt.utcs/lons/lats` at the moment. Refactoring that would quite some work I think